### PR TITLE
Add optional dft_length input to the DFT and IDFT operators.

### PR DESCRIPTION
**Pull request by smk2007**
Add optional dft_length input to the DFT and IDFT operators.

### CODE REVIEW

File: onnxruntime/contrib_ops/cpu/signal/dft.cc [hunk 1]
```diff
@@ -324,26 +342,35 @@ static Status discrete_fourier_transform(OpKernelContext* ctx, const Tensor* X,
...
 
 static Status discrete_fourier_transform(OpKernelContext* ctx, int64_t axis, bool is_onesided, bool inverse) {
   // Get input shape
   const auto* X = ctx->Input<Tensor>(0);
+  const auto* dft_length = ctx->Input<Tensor>(1);
   const auto& X_shape = X->Shape();
   const auto is_real_valued = is_real_valued_signal(X_shape);
   const auto is_complex_valued = is_complex_valued_signal(X_shape);
 
+  int64_t shape_number_of_samples = static_cast<int64_t>(X_shape[axis]);
```
Comment (martinb35) [inline onnxruntime/contrib_ops/cpu/signal/dft.cc:363]: is shape_number_of_samples unused?

File: onnxruntime/contrib_ops/cpu/signal/dft.cc [hunk 2]
```diff
@@ -242,60 +262,58 @@ static Status dft_naive(const Tensor* X, Tensor* Y,
...
     std::complex<T>& out = *(Y_data + i*Y_stride);
     out.real(0);
     out.imag(0);
 
-    for (size_t j = 0; j < number_of_samples; j++) {  // vectorize over this loop
+    for (size_t j = 0; j < dft_length; j++) {  // vectorize over this loop
       auto exponential = std::complex<T>(cos(i * j * angular_velocity), sin(i * j * angular_velocity));
       auto window_element = window_data ? * (window_data + j) : 1;
-      auto element = *(X_data + j*X_stride) * window_element;
+      auto x = (j < number_of_samples) ? *(X_data + j * X_stride) : 0;
```
Comment (martinb35) [inline onnxruntime/contrib_ops/cpu/signal/dft.cc:281]: it looks like number_of_samples was replaced with dft_length almost everywhere except a few spots - are the spots where it's still being used correct? I haven't dug into the math, just asking.

File: onnxruntime/contrib_ops/cpu/signal/dft.cc (hunk 1 above)
Comment (smk2007) [reply onnxruntime/contrib_ops/cpu/signal/dft.cc:363]: no removed

File: onnxruntime/contrib_ops/cpu/signal/dft.cc (hunk 2 above)
Comment (smk2007) [reply onnxruntime/contrib_ops/cpu/signal/dft.cc:281]: Yeah, a condition needed to be added for when the dft_length > the actual number of samples available in the tensor

**Pull request merged**